lib/__tests__/test.js

```diff
@@ -2,7 +2,45 @@
 
 const fs = require('fs');
 const path = require('path');
-const defineTest = require('jscodeshift/src/testUtils').defineTest;
+const {walkDir} = require('../execution-helper');
+
+/**
+ * @param {string} dirname - the testing directory
+ * @param {string} transformName - the name of the transform
+ * @param {string} filePrefix - the test file prefix
+ */
+function defineTest(dirname, transformName, filePrefix) {
+  const testName = `rewrites correctly on ${filePrefix}`;
+  describe(transformName, () => {
+    it(testName, () => {
+      runTest(dirname, transformName, filePrefix);
+    });
+  });
+}
+
+/**
+ * @param {string} dirname - the testing directory
+ * @param {string} transformName - the name of the transform
+ * @param {string} filePrefix - the test file prefix
+ */
+function runTest(dirname, transformName, filePrefix) {
+  const fixtureDir = path.join(dirname, '..', '__testfixtures__');
+  const inputPath = path.join(fixtureDir, filePrefix + '.input.js');
+  const source = fs.readFileSync(inputPath, 'utf8');
+  const expectedOutput = fs.readFileSync(
+      path.join(fixtureDir, filePrefix + '.output.js'),
+      'utf8'
+  );
+
+  // Assumes transform is one level up from __tests__ directory
+  const transform = require(path.join(dirname, '..', transformName));
+  const rewriterOutput = transform({
+    path: inputPath,
+    source,
+  });
+
+  expect((rewriterOutput || '').trim()).toEqual(expectedOutput.trim());
```

Comment:
Does this work without `trim()` or the `|| ''`? That'd be cleaner if we can manage it.

Reply:
The `|| ''` I can take out no problem but the testing is getting mad at me about having trailing spaces in my files. Alternatives are remove the trailing spaces, or remove the trim.

Reply:
I'm a bit torn here, but it's probably cleaner to remove the trailing spaces. We should ideally be testing against the exact output, and having our fixture files represent the exact output.

This may make editing the files tricky if you (like me) have your editor configured to automatically add trailing newlines to files. The best way to get around this is with [EditorConfig](https://editorconfig.org/). In particular, install the [VSCode EditorConfig plugin](https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig), and then add (either in this PR or a separate one) an .editorconfig file.

A good base .editorconfig file to use is something like https://github.com/domenic/worm-scraper/blob/master/.editorconfig . I.e., set reasonable defaults for the whole project, then have exceptions for specific directories.

Once this is all setup, VS Code---and many other editors---would read "whether to add a final newline or not" from the .editorconfig file, and thus do the right thing, adding it for most files, but not messing with the fixture outputs.

Reply:
Can we just make the rewriter put a newline at the end?